e2e/testdata/fn-eval/resource-deletion/config.yaml

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: delete-all
+source: |
+  # delete all resources
+  ctx.resource_list["items"] = ctx.resource_list["items"][:1]
```

Comment:
why can't we assign it to empty list `[]` ?

Reply:
If all resources are removed than `fn eval` will not modify the resource. https://github.com/GoogleContainerTools/kpt/issues/1460

Reply:
hmm. I think we need to change that behavior to make it consistent. Let's prioritize #1460